daemon/cmd/daemon.go

```diff
@@ -1065,13 +1059,29 @@ func newDaemon(ctx context.Context, cleaner *daemonCleanup, params *daemonParams
 	// router IPs from the K8s resources if we weren't able to restore them
 	// from the fs. We must do this after IPAM because we must wait until the
 	// K8s resources have been synced. Part 2/2 of restoration.
-	if option.Config.EnableIPv4 {
-		d.restoreCiliumHostIPs(false, router4FromK8s)
-	}
-	if option.Config.EnableIPv6 {
-		d.restoreCiliumHostIPs(true, router6FromK8s)
+	bo := wait.Backoff{
+		Duration: 100 * time.Millisecond,
+		Factor:   1,
+		Jitter:   0.1,
+		Steps:    3,
+	}
+	err = wait.ExponentialBackoffWithContext(ctx, bo, func(ctx context.Context) (done bool, err error) {
+		var errs error
+		if option.Config.EnableIPv4 {
+			errs = errors.Join(errs, d.restoreCiliumHostIPs(false, router4FromK8s))
+		}
+		if option.Config.EnableIPv6 {
+			errs = errors.Join(errs, d.restoreCiliumHostIPs(true, router6FromK8s))
+		}
+		if resiliency.IsRetryable(errs) {
+			log.WithError(errs).Warnf("Failed to remove old router IPs from cilium_host. Retrying...")
+			return false, nil
+		}
+		return true, errs
+	})
+	if err != nil {
+		log.WithError(err).Error("Restore host ips failed. Manual intervention is required to remove all other old IPs.")
```

Comment:
```suggestion
		log.WithError(err).Error("Restore of the cilium_host IP failed. Manual intervention is required to remove all other old IPs.")
```

Just to not differ too much from the original log msg.

Reply:
Good catch! Tx Chris!!